opm/autodiff/ExtendedBlackoilModel.hpp

```diff
@@ -0,0 +1,286 @@
+/*
+  Copyright 2015 IRIS AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_EXTENDEDBLACKOILMODEL_HEADER_INCLUDED
+#define OPM_EXTENDEDBLACKOILMODEL_HEADER_INCLUDED
+
+#include <opm/autodiff/BlackoilModelBase.hpp>
+#include <opm/autodiff/BlackoilModelParameters.hpp>
+#include <opm/autodiff/ExtendedBlackoilState.hpp>
+#include <opm/autodiff/WellStateFullyImplicitBlackoilSolvent.hpp>
+#include <opm/autodiff/SolventPropsAdFromDeck.hpp>
+
+namespace Opm {
+
+    /// A model implementation for three-phase black oil
+    /// with one extra component.
+    ///
+    ///
+    /// It uses automatic differentiation via the class AutoDiffBlock
+    /// to simplify assembly of the jacobian matrix.
+    template<class Grid>
+    class ExtendedBlackoilModel : public BlackoilModelBase<Grid, ExtendedBlackoilModel<Grid> >
```

Comment:
I would prefer if this class was called BlackoilSolventModel. The polymer model is called BlackoilPolymerModel, so it is more consistent with that, and it also communicates better that it is in fact a solvent model and not an arbitrary extension.